src/index.test.tsx

```diff
@@ -204,3 +204,215 @@ test('changing subscribable instances', async () => {
+  expect(div.textContent).to.equal('string, b');
+
+  const rep = new FakeReplicache();
+
+  render(<A key="c" rep={rep} def="c" />, div);
+  expect(div.textContent).to.equal('string, c');
+  await promise;
+  await sleep(1);
+  expect(div.textContent).to.equal('bigint, 123');
+});
+
+test.skip('using isEqual [type checking]', async () => {
+  const use = (...args: unknown[]) => args;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function expectType<T>(_expression: T) {
+    // intentionally empty
+  }
+
+  class WithIsEqual<Tx> implements SubscribableWithIsEqual<Tx> {
+    subscribe<Data>(
+      query: (tx: Tx) => Promise<Data>,
+      options: {
+        onData: (data: Data) => void;
+        isEqual?: ((a: Data, b: Data) => boolean) | undefined;
+      },
+    ): () => void {
+      use(query, options);
+      return () => undefined;
+    }
+  }
+
+  {
+    const s = useSubscribe(
+      new WithIsEqual<ReadTransaction>(),
+      tx => {
+        use(tx);
+        return Promise.resolve(123n);
+      },
+      {isEqual: (a, b) => a === b},
+    );
+    expectType<bigint | undefined>(s);
+  }
+
+  {
+    // This is invalid because the return type is not json
+    // @ts-expect-error index.ts(61, 3): An argument for 'options' was not provided.
+    const s = useSubscribe(new WithIsEqual<ReadTransaction>(), tx => {
+      use(tx);
+      return Promise.resolve(123n);
+    });
+    use(s);
+  }
+
+  {
+    // default not passed so it is undefined
+    const s = useSubscribe(new WithIsEqual<ReadTransaction>(), tx => {
+      use(tx);
+      return Promise.resolve(123);
+    });
+    expectType<123 | undefined>(s);
+  }
+
+  {
+    const s = useSubscribe(
+      new WithIsEqual<ReadTransaction>(),
+      tx => {
+        use(tx);
+        return Promise.resolve(123);
+      },
+      456,
+    );
+    expectType<123 | 456>(s);
+  }
+
+  {
+    const s: bigint | 'abc' = useSubscribe(
+      new WithIsEqual<ReadTransaction>(),
+      tx => {
+        use(tx);
+        return Promise.resolve(123n);
+      },
+      {isEqual: (a, b) => a === b, default: 'abc'},
+    );
+    expectType<bigint | 'abc'>(s);
+  }
+});
+
+test('No isEqual [type checking]', async () => {
+  const use = (...args: unknown[]) => args;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function expectType<T>(_expression: T) {
+    // intentionally empty
+  }
+
+  class WithNoIsEqual<Tx> implements Subscribable<Tx> {
+    subscribe<Data extends ReadonlyJSONValue | undefined>(
+      query: (tx: Tx) => Promise<Data>,
+      {onData}: {onData: (data: Data) => void},
+    ): () => void {
+      use(query, onData);
+      return () => undefined;
+    }
+  }
+
+  {
+    const s = useSubscribe(new WithNoIsEqual<ReadTransaction>(), tx => {
+      use(tx);
+      return Promise.resolve(123);
+    });
+    expectType<123 | undefined>(s);
+  }
+
+  {
+    // When subscribe does not support isEqual, the options is treated as default :'(
```

Comment:
This is pretty unfortunate.

Maybe we should do a major release and not support all this crazy overloads?

Reply:
Yes, let's just do:

```
useSubscribe(r, query, opts?)
```

Where opts is `def`, `isEqual`, and `deps`.

Reply:
I prefer the name `default` over `def`. I'm fine with `deps` because `dependencies` is so long but maybe it would be better to be more clear... especially if we consider dependencies to be very rarely used. I guess that convinced me. I will update the PR with the major breaking changes.